packages/shared-ux/chrome/navigation/src/ui/components/navigation_section_ui.tsx

```diff
@@ -558,22 +556,21 @@ export const NavigationSectionUI: FC<Props> = React.memo(({ navNode: _navNode })
     ): EuiCollapsibleNavSubItemProps[] | undefined => {
       if (!_items) return;
 
-      return _items.map((item) => {
+      return _items.map(({ path, ...item }) => {
         if (item.renderItem) {
           return item;
         }
-        // @ts-ignore - TODO
-        const parsed: EuiCollapsibleNavSubItemProps = {
+        const parsed = {
           ...item,
           items: serializeAccordionItems(item.items),
           accordionProps:
             item.items !== undefined
-              ? getAccordionProps(item.path ?? item.id!, {
+              ? getAccordionProps(path ?? item.id!, {
                   onClick: item.onClick,
                   ...item.accordionProps,
                 })
               : undefined,
-        };
+        } as EuiCollapsibleNavSubItemProps;
```

Comment:
I try to avoid casting as much as possible. I know that this is more verbose but at least it is explicit what the expected interface is.

```ts
return _items.map(({ path, ...item }) => {
  if (item.renderItem) {
    return item;
  }

  const itemsSerialized: EuiCollapsibleNavSubItemProps['items'] = serializeAccordionItems(
    item.items
  );

  const accordionProps =
    itemsSerialized === undefined
      ? undefined
      : getAccordionProps(path ?? item.id!, {
          onClick: item.onClick,
          ...item.accordionProps,
        });

  // The EuiCollapsibleNavSubItemProps can have *either* href/linkProps or items/accordionProps/isCollapsible. Mixing them is not allowed
  // See ExclusiveUnion type in EUI repo at eui/src/components/collapsible_nav_beta/collapsible_nav_item/collapsible_nav_item.tsx#L63-L102
  const {
    href,
    linkProps,
    isCollapsible,
    items: __items,
    accordionProps: __accordionProps,
    ...rest
  } = item;

  if (itemsSerialized) {
    const parsed: EuiCollapsibleNavSubItemProps = {
      ...rest,
      items: itemsSerialized,
      accordionProps,
      isCollapsible,
    };

    return parsed;
  }

  const parsed: EuiCollapsibleNavSubItemProps = {
    ...rest,
    href,
    linkProps,
  };

  return parsed;
});
```

Reply:
It's casting because it has to be - see the commit message:

>  Force cast the remaining `@ts-ignore`
> - Not sure there's anything easy to do here, the `path` key is custom compared to EUI so there's naturally going to be mismatch from what the function outputs to what the component accepts

The types will not work without it because the props you're giving it (with the custom `path` prop) does not match what the EUI component expects.